client/elasticClientCommon.go

```diff
@@ -102,32 +102,68 @@ func elasticBulkRequestResponseHandler(res *esapi.Response) error {
 		return fmt.Errorf("%w cannot read elastic response body bytes", err)
 	}
 
-	bulkResponse := &BulkRequestResponse{}
-	err = json.Unmarshal(bodyBytes, bulkResponse)
+	return extractErrorFromBulkBodyResponseBytes(bodyBytes, index)
+}
+
+func extractErrorFromBulkBodyResponseBytes(bodyBytes []byte, index string) error {
+	if index == indexer.TransactionsIndex {
```

Comment:
could have used polymorphism instead of this ugly if

Reply:
refactored not need.